File: mpp/linked-list-set-maxim092001/src/test/java/linked_list_set/LinearizabilityTest.java
package linked_list_set;

import org.jetbrains.kotlinx.lincheck.LinChecker;
import org.jetbrains.kotlinx.lincheck.annotations.Operation;
import org.jetbrains.kotlinx.lincheck.annotations.Param;
import org.jetbrains.kotlinx.lincheck.paramgen.IntGen;
import org.jetbrains.kotlinx.lincheck.strategy.stress.StressCTest;
import org.junit.Test;

@StressCTest(sequentialSpecification = SequentialSetImpl.class)
@Param(name = "key", gen = IntGen.class, conf = "1:5")
public class LinearizabilityTest {
    private Set set = new SetImpl();

    @Operation(params = "key")
    public boolean add(int x) {
        return set.add(x);
    }

    @Operation(params = "key")
    public boolean contains(int x) {
        return set.contains(x);
    }

    @Operation(params = "key")
    public boolean remove(int x) {
        return set.remove(x);
    }

    @Test
    public void test() {
        LinChecker.check(LinearizabilityTest.class);
    }
}

File: mpp/fine-grained-bank-maxim092001/test/LinearizabilityTest.kt
import org.jetbrains.kotlinx.lincheck.LinChecker
import org.jetbrains.kotlinx.lincheck.annotations.Operation
import org.jetbrains.kotlinx.lincheck.annotations.Param
import org.jetbrains.kotlinx.lincheck.paramgen.IntGen
import org.jetbrains.kotlinx.lincheck.paramgen.LongGen
import org.jetbrains.kotlinx.lincheck.strategy.stress.StressCTest
import org.jetbrains.kotlinx.lincheck.verifier.VerifierState
import org.junit.Test
import java.util.*

/**
 * This test checks bank implementation for linearizability.
 */
@Param.Params(
    Param(name = "id", gen = IntGen::class, conf = "0:4"),
    Param(name = "amount", gen = LongGen::class, conf = "1:100")
)
@StressCTest
class LinearizabilityTest : VerifierState() {
    private val bank: Bank = BankImpl(5)

    @Operation(params = ["id"])
    fun getAmount(id: Int): Long =
        bank.getAmount(id)

    @get:Operation
    val totalAmount: Long
        get() = bank.totalAmount

    @Operation(params = ["id", "amount"], handleExceptionsAsResult = [IllegalStateException::class])
    fun deposit(id: Int, amount: Long): Long =
        bank.deposit(id, amount)

    @Operation(params = ["id", "amount"], handleExceptionsAsResult = [IllegalStateException::class])
    fun withdraw(id: Int, amount: Long): Long =
        bank.withdraw(id, amount)

    @Operation(params = ["id", "id", "amount"], handleExceptionsAsResult = [IllegalStateException::class])
    fun transfer(idFrom: Int, idTo: Int, amount: Long) {
        if (idFrom != idTo) bank.transfer(idFrom, idTo, amount)
    }

    override fun extractState(): Any {
        val amounts: MutableList<Long> = ArrayList()
        for (i in 0 until bank.numberOfAccounts) amounts.add(bank.getAmount(i))
        return amounts
    }

    @Test
    fun test() {
        LinChecker.check(LinearizabilityTest::class.java)
    }
}